src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -401,10 +420,11 @@ func (q *qemuArchBase) appendBlockImage(ctx context.Context, devices []govmmQemu
 	return devices, nil
 }
 
-func genericSCSIController(enableIOThreads, nestedRun bool) (govmmQemu.SCSIController, *govmmQemu.IOThread) {
+func genericSCSIController(enableIOThreads, nestedRun bool, memEncrypt bool) (govmmQemu.SCSIController, *govmmQemu.IOThread) {
 	scsiController := govmmQemu.SCSIController{
 		ID:            scsiControllerID,
 		DisableModern: nestedRun,
+		IOMMUPlatform: memEncrypt,
```

Comment:
Doesn't the govmm code handle this already?  In any case it shouldn't be necessary if you use `confidential-guest-support=`.